File: livingrimoire start here/LivinGrimoire java/src/Skills/special_skills/GamificationP.java
package Skills.special_skills;

import Auxiliary_Modules.AXGamification;
import LivinGrimoire.DiSkillV2;
import LivinGrimoire.Neuron;

public class GamificationP extends DiSkillV2 {
    // the grind side of the game, see GamificationN for the reward side
    private int gain = 1;
    private DiSkillV2 skill;
    private AXGamification axGamification = new AXGamification();

    public GamificationP(DiSkillV2 skill) {
        this.skill = skill;
    }

    public void setGain(int gain) {
        if (gain >0){
            this.gain = gain;}
    }

    public AXGamification getAxGamification() {
        // shallow ref
        return axGamification;
    }

    @Override
    public void input(String ear, String skin, String eye) {
        skill.input(ear, skin, eye);
    }

    @Override
    public void output(Neuron noiron) {
        // skill activation increases gaming credits
        if (skill.pendingAlgorithm()) {axGamification.incrementBy(gain);}
        skill.output(noiron);
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/skills/special/GamificationP.kt
package skills.special

import auxiliary_modules.AXGamification
import livinGrimoire.DiSkillV2
import livinGrimoire.Neuron

class GamificationP(skill: DiSkillV2) : DiSkillV2() {
    // the grind side of the game, see GamificationN for the reward side
    private var gain = 1
    private val skill: DiSkillV2
    private val axGamification: AXGamification = AXGamification()

    init {
        this.skill = skill
    }

    fun setGain(gain: Int) {
        if (gain > 0) {
            this.gain = gain
        }
    }

    fun getAxGamification(): AXGamification {
        // shallow ref
        return axGamification
    }

    override fun input(ear: String, skin: String, eye: String) {
        skill.input(ear, skin, eye)
    }

    override fun output(noiron: Neuron) {
        // skill activation increases gaming credits
        if (skill.pendingAlgorithm()) {
            axGamification.incrementBy(gain)
        }
        skill.output(noiron)
    }
}